mongo/options/example_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package options_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type CustomLogger struct {
+	io.Writer
+	mu sync.Mutex
+}
+
+func (logger *CustomLogger) Info(level int, msg string, _ ...interface{}) {
+	logger.mu.Lock()
+	defer logger.mu.Unlock()
+
+	logger.Write([]byte(fmt.Sprintf("level=%d msg=%s\n", level, msg)))
+}
+
+func (logger *CustomLogger) Error(err error, msg string, _ ...interface{}) {
+	logger.mu.Lock()
+	defer logger.mu.Unlock()
+
+	logger.Write([]byte(fmt.Sprintf("err=%v msg=%s\n", err, msg)))
+}
+
+func ExampleClientOptions_SetLoggerOptions() {
```

Comment:
Optional: This example test will create an "Example" section in the "SetLoggerOptions" method documentation. Consider adding `_customLogger` to the end of the example test name so the example section renders as "Example (CustomLogger)" instead to indicate it's an example of how to write and use a custom logger.

```suggestion
func ExampleClientOptions_SetLoggerOptions_customLogger() {
```